src/reflect/value.go

```diff
@@ -432,6 +432,33 @@ func (v Value) Slice(i, j int) Value {
 }
 
 func (v Value) Slice3(i, j, k int) Value {
+	switch v.Kind() {
+	case Slice:
+		hdr := *(*sliceHeader)(v.value)
+		i, j, k := uintptr(i), uintptr(j), uintptr(k)
+
+		if i < 0 || i > hdr.len || j < 0 || j > hdr.len || k < 0 || k > hdr.len {
```

Comment:
```suggestion
		if i < 0 || j < i || k < j || hdr.len < k {
```

But because `i` is an `uintptr`, `i < 0` doesn't do anything so can be left out:

```suggestion
		if j < i || k < j || hdr.len < k {
```

This is in fact what the compiler does. It's correct because: if `k` is negative, it wraps around and will be bigger than `hdr.len`. Similarly, if `j` is negative (but `k` is in bounds), `j` wraps around and is thus larger than `k`. And so on for `i`.